test/sys/test_event.rs

```diff
@@ -0,0 +1,19 @@
+use std::time::Duration;
+
+use libc::timespec;
+use nix::sys::event::{kqueue, kevent};
+
+#[test]
+pub fn test_kevent() {
+    let kq = kqueue().unwrap();
+
+    let mut events = Vec::new();
+
+    let timeout = Duration::from_millis(0);
+    assert_eq!(kevent(kq, &[], &mut events, Some(timeout)).unwrap(), 0);
+
+    let timeout = timespec { tv_sec: 0, tv_nsec: 0};
+    assert_eq!(kevent(kq, &[], &mut events, Some(timeout)).unwrap(), 0);
+
+    assert_eq!(kevent::<timespec>(kq, &[], &mut events, None).unwrap(), 0);
```

Comment:
This is the unfortunate side effect of these option types. I'd suggest having these all as doctests instead of integration tests, since these examples will be helpful for users.